supyr_struct/buffer.py

```diff
@@ -106,27 +110,56 @@ class Buffer():
     Buffers are simply a wrapper around another object which
     gives it an interface that mimics the read, seek, size,
     tell, and write methods found in mmap and file objects.
+
     Buffers also implement a peek function for reading the next
     X number of bytes without changing the read/write position.
     '''
+    # Slots should have '_pos' but this creates a C level conflict when trying
+    # to combined inherit this class with the internal bytes class.
     __slots__ = ()
 
+    def __init__(self, *args):
+        # Dummy __init__ that makes sure there is always a self._pos.
+        # Accepts args like *args to account for child objects.
+        self._pos = 0
+
     def read(self, count=None):
+        '''
+        read stub. Meant for overloading.
+
+        Should increment self._pos by the number of bytes succesfully read.
+        And return those bytes.
+        '''
         raise NotImplementedError('read method must be overloaded.')
 
     def seek(self, pos, whence=SEEK_SET):
+        '''
+        seek stub. Meant for overloading.
+
+        Should implement correct handing for SEEK_SET, SEEK_CUR, SEEK_END.
+        Setting self._pos absolutely, adding pos to self._pos, and setting
+        self._pos to the maximum value for this buffer.
+        '''
         raise NotImplementedError('seek method must be overloaded.')
 
     def size(self):
+        '''
+        Get the size of this buffer.
+        '''
         return len(self)
 
     def tell(self):
+        '''
+        tell stub. Meant for overloading.
+
+        Should tell the caller what the current value of self._pos is.
+        '''
         raise NotImplementedError('tell method must be overloaded.')
 
     def peek(self, count=None, offset=None):
         '''
         Reads and returns 'count' number of bytes from the Buffer
-        without changing the current read/write pointer position.
+        without changing the value of self._pos.
```

Comment:
The user doesn't need to know about implementation details like _pos because they shouldn't know what they do. The old wording was a bit wonky though, so i guess just change it to whatever works.